pycloudlib/ibm_softlayer/instance.py

```diff
@@ -0,0 +1,374 @@
+
+        Args:
+            vs_manager: Softlayer VSManager (Virtual Server Manager) instance
+            target_image_global_identifier: image global identifier
+            hostname: instance hostname
+            flavor: instance flavor
+            datacenter: datacenter region
+            public_security_group_ids: list of public security group ids
+            private_security_group_ids: list of private security group ids
+            ssh_key_ids: list of ssh key ids
+            domain_name: domain name
+
+        """
+        logger.debug("Creating raw instance")
+        constant_args = {
+            "private": False,
+            "dedicated": False,  # default
+            "hourly": True,  # default
+            "local_disk": False,  # default
+        }
+        instance_specific_args = {
+            "domain": domain_name,
+            "hostname": hostname,
+            "ssh_keys": ssh_key_ids,
+            "image_id": target_image_global_identifier,
+            "flavor": flavor,
+            "datacenter": datacenter,
+            "public_security_groups": public_security_group_ids,
+            "private_security_groups": private_security_group_ids,
+        }
+        # check if instance configuration is valid
+        try:
+            logger.info(
+                "Verifying configuration for instance before creating it."
+            )
+            vs_manager.verify_create_instance(
+                **constant_args,
+                **instance_specific_args,
+                **kwargs,
+            )
+        except SoftLayer.SoftLayerAPIError as e:
+            logger.error("configuration for instance is invalid: %s", e)
+            raise IBMSoftlayerException(
+                f"Failed to verify instance configuration: {e}"
+            ) from e
+        except Exception as e:
+            logger.error(
+                "Unexpected error while verifying instance configuration: %s",
+                e,
+            )
+            raise IBMSoftlayerException(
+                f"Unexpected error while verifying instance configuration: {e}"
+            ) from e
+        else:
+            logger.info(
+                "Configuration for instance is valid. Creating instance now."
+            )
+            raw_instance = vs_manager.create_instance(
+                **constant_args,
+                **instance_specific_args,
+                **kwargs,
+            )
+            logger.info("Created instance %s", raw_instance["hostname"])
+            full_raw_instance_info = vs_manager.get_instance(
+                raw_instance["id"]
+            )
+            logger.debug("New instance details: %s", full_raw_instance_info)
+            return full_raw_instance_info
+
+    def get_boot_id(self):
+        """Get the instance boot_id.
+
+        Returns:
+            string with the boot UUID
+        """
+        result = self.execute(
+            "cat /proc/sys/kernel/random/boot_id", no_log=True
```

Comment:
Why no_log? Is the boot_id sensitive info? If so, cloud we leave a breadcrumb comment explaining it?

Reply:
okay so apparently, i just copied this from the BaseInstance class thinking that i would need to modify it but never
ended up needing to. And so the no log is default behavior for all clouds apparently. So i just removed the duplicate
code from the IBMSoftlayerInstance Class.